docker/setup_trust_store.sh

```diff
@@ -64,9 +64,9 @@ setup_auth_server_trust_store() {
     -trustcacerts \
     -noprompt \
     -alias "uaa_ca" \
-    -file "${UAA_CA_PATH}" \
+    -file "${UAA_CA_PATH:?}" \
```

Comment:
Why are you adding `:?` to all variable lookups? It seems like it has the exact same effect as `set -u` (`--nounset`), which is present at the beginning of the script.

Reply:
It is redundant with `set -u`, except that shellcheck wants some reassurance that those variables are all initialized already and doesn't accept `set -u` as a sufficient solution.

I would have preferred to use a declare statement instead, basically declaring at the top of the script what variables it's depending on being already set in the environment. But that also has the side effect of giving them a default value and circumvents the benefit of the `set -u`.

Do you have a better suggestion for making shellcheck happy in the case where a variable is not initialized in the script where it's referenced?

Reply:
It would a better bug report if you had used "--enable=check-unassigned-uppercase" instead of "--enable=all". It looks like a doc problem to me.

Reply:
I have removed all the added ":?" notations so we can get this PR merged.